src/main/java/com/jcraft/jsch/Slf4jLogger.java

```diff
@@ -1,49 +1,58 @@
 package com.jcraft.jsch;
 
+import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.slf4j.event.Level;
+import org.slf4j.spi.LoggingEventBuilder;
 
 public class Slf4jLogger implements com.jcraft.jsch.Logger {
 
-  private static final org.slf4j.Logger logger = LoggerFactory.getLogger(JSch.class);
+  private static final Logger stlogger = LoggerFactory.getLogger(JSch.class);
+  private Logger logger;
 
-  public Slf4jLogger() {}
+  public Slf4jLogger() {
+    this(stlogger);
+  }
+
+  Slf4jLogger(Logger logger) {
+    this.logger = logger;
+  }
 
   @Override
   public boolean isEnabled(int level) {
-    switch (level) {
-      case com.jcraft.jsch.Logger.DEBUG:
-        return logger.isDebugEnabled();
-      case com.jcraft.jsch.Logger.INFO:
-        return logger.isInfoEnabled();
-      case com.jcraft.jsch.Logger.WARN:
-        return logger.isWarnEnabled();
-      case com.jcraft.jsch.Logger.ERROR:
-      case com.jcraft.jsch.Logger.FATAL:
-        return logger.isErrorEnabled();
-      default:
-        return logger.isTraceEnabled();
-    }
+    return logger.isEnabledForLevel(getLevel(level));
```

Comment:
I think we need to revert this part: I believe the `isEnabledForLevel()` method is only available with SLFJ4 2.x, but we should keep compatibility with the SLF4J 1.x API as well.

Reply:
I assumed that slf4j 2.0 is used because it's in the pom.xml

Reply:
> I assumed that slf4j 2.0 is used because it's in the pom.xml

I switched the dependency to 2.x because the 1.x versions aren't modularized, so they caused warnings for `module-info.java` for JPMS support.
But I made sure that the `Slf4jLogger` was fully compatible with 1.x, allowing users to decide which version they want to use.

Reply:
If backward compatibility to slf4j 1 is intended, the pom should be changed to use that library to make sure that the compiler always compiles against the correct methods.

Reply:
> If backward compatibility to slf4j 1 is intended, the pom should be changed to use that library to make sure that the compiler always compiles against the correct methods.

Except the project doesn't compile `module-info.java` correctly if it is built with the 1.x version of the library, and then we lose JPMS support.

It doesn't seem that difficult to just leave the code as it is with the existing `switch` statement (it works with both SLFJ4 1.x & 2.x), and the way I structured the project, users can make use of either version in their own projects.

Reply:
I'll make a break, then will do the change back to v1-compatibility

Reply:
> Except the project doesn't compile `module-info.java` correctly if it is built with the 1.x version of the library, and then we lose JPMS support.

OK. I've just been bitten once already when compiling against a newer API while executing with an older one. It's been a while but a lecture for life ;-) In the good old times when a kilobyte weighted a kilogram, `java.lang.BigDecimal` got a new constructor `BigDecimal(long)`. While I don't expect something like that took place in Slf4J, it's still dangerous to do so.

Reply:
Yep, I realize that and I wouldn't have changed the pom.xml to the 2.x version, except that I was also trying to make JSch fully modularized & JPMS compatible, and this was the only way I found to accomplish that.
I think as long as we are careful about what SLF4J methods we utilize, we can accomplish everything:
1. SLF4j 1.x compability.
2. SLF4j 2.x compability.
3. Java 9+ JPMS support.

Reply:
Hm, I'm not that Maven-savvy but isn't it possible to define a test-only dependency. Can we use slf4j 2 for compiling and slf4j 1 for testing? With the test cases I've added any incompatibility (and be it compiler-related) should show up during test that way.

Reply:
I think the only way to do that would be to completely restructure as a multi-module project, with the library as one module and the tests as a different module, but that would be alot of work.

Reply:
I've changed the implementation to only use the API of slf4j 1